waypoint/services/nats_service.py

```diff
@@ -15,6 +18,15 @@
 logger = get_logger(__name__)
 
 
+def retry_log(retry_state: RetryCallState):
+    """Custom logging for retry attempts."""
+    if retry_state.outcome.failed:
+        exception = retry_state.outcome.exception()
+        logger.warning(
+            f"Retry attempt {retry_state.attempt_number} failed due to {type(exception).__name__}: {exception}"
+        )
+
+
```

Comment:
This can be made a private method of the NatsEventProcessor class

Also the log line can use the {}-interpolation pattern